front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/V2PipelineTemplateController.java

```diff
@@ -145,10 +172,34 @@ private void checkForDuplicatePipelineTemplate(String id) {
     throw new DuplicateEntityException("A pipeline template with the id " + id + " already exists");
   }
 
+  private String computeSHA256Digest(PipelineTemplate pipelineTemplate) {
+    // Sorted RB Tree by keys.
+    TreeMap<String, Object> sortedMap = new TreeMap<>(pipelineTemplate);
+    try {
+      String jsonPayload = objectMapper.writeValueAsString(sortedMap).replaceAll("\\s+", "");
+      MessageDigest digest = MessageDigest.getInstance("SHA-256");
+      byte[] hashBytes = digest.digest(jsonPayload.getBytes(StandardCharsets.UTF_8));
+      return Hex.encodeHexString(hashBytes);
+    } catch (NoSuchAlgorithmException | JsonProcessingException e) {
+      throw new InvalidRequestException(String.format("Computing digest for pipeline template %s failed. Nested exception is %s", pipelineTemplate.getId(), e));
+    }
+  }
+
   private PipelineTemplateDAO getPipelineTemplateDAO() {
     if (pipelineTemplateDAO == null) {
       throw new BadRequestException("Pipeline Templates are not supported with your current storage backend");
     }
     return pipelineTemplateDAO;
   }
+
+  private void validatePipelineTemplateVersion(String version) {
+    if (StringUtils.isEmpty(version)) {
+      throw new InvalidRequestException("The provided version is empty or null");
```

Comment:
I thought empty was OK, and it would point to latest?

Reply:
Yep.